operator-wg/whitepaper/051_kubebuilder.md

```diff
@@ -0,0 +1,17 @@
+## kubebuilder Operator Framework
+
+The kubebuilder framework provides developers the possibilities to extend the Kubernetes API by using Custom Resource Definitions, and to create operators that handle these custom resources.
+
+The main concept provided by the kubebuilder framework is a *Manager*. In the same way the native Kubernetes controllers are grouped into a single Kubernetes Controller Manager (`kube-controller-manager`), you will be able to create several operators and make them managed by a single manager.
+
+As Kubernetes API resources are attached to domains and arranged in Groups, Versions and Kinds, the Kubernetes custom resources you will define will be attached to your own domain, and arranged in your own groups, versions and kinds.
+
+The first step when using kubebuilder is to create a project, that  will create the source code for building a single Manager attached to your domain.
+
+After initiating your project with a specific domain, you can add APIs to your domain and make these APIs managed by the manager.
+
+Adding a resource to the project will generate some sample code for you: a sample *Custom Resource Definition* that you will adapt to build your own custom resource, and a sample *Reconciler* that will implement the reconcile loop for your operator handling this resource.
+
+The kubebuilder framework leverages the `controller-runtime` library, that provides the Manager and Reconciler concepts, among others.
```

Comment:
Should also mention controller-tools which provides the code gen.

Reply:
While technically speaking controller-tools is part of the kubebuilder ecosystem, its relevance to the end user is lower as it is abstracted/hidden behind a make target.